#!/usr/bin/env python
PACKAGE = "phobos_simulation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("link0_Kp", double_t, 0, "link0_Kp", 1.0, 0, 10)
gen.add("link0_Ki", double_t, 0, "link0_Ki", 0.2, 0, 10)
gen.add("link0_Kd", double_t, 0, "link0_Kd", 0.8, 0, 10)

gen.add("link1_Kp", double_t, 0, "link1_Kp", 1.0, 0, 10)
gen.add("link1_Ki", double_t, 0, "link1_Ki", 0.2, 0, 10)
gen.add("link1_Kd", double_t, 0, "link1_Kd", 0.8, 0, 10)

gen.add("link2_Kp", double_t, 0, "link2_Kp", 1.0, 0, 10)
gen.add("link2_Ki", double_t, 0, "link2_Ki", 0.2, 0, 10)
gen.add("link2_Kd", double_t, 0, "link2_Kd", 0.8, 0, 10)

gen.add("link3_Kp", double_t, 0, "link3_Kp", 1.0, 0, 10)
gen.add("link3_Ki", double_t, 0, "link3_Ki", 0.2, 0, 10)
gen.add("link3_Kd", double_t, 0, "link3_Kd", 0.8, 0, 10)

gen.add("link4_Kp", double_t, 0, "link4_Kp", 1.0, 0, 10)
gen.add("link4_Ki", double_t, 0, "link4_Ki", 0.2, 0, 10)
gen.add("link4_Kd", double_t, 0, "link4_Kd", 0.6, 0, 10)

gen.add("finger_Kp", double_t, 0, "finger_Kp", 1.0, 0, 10)
gen.add("finger_Ki", double_t, 0, "finger_Ki", 0.5, 0, 10)
gen.add("finger_Kd", double_t, 0, "finger_Kd", 0.6, 0, 10)


exit(gen.generate(PACKAGE, "phobos_simulation", "ArmJointDRC"))
